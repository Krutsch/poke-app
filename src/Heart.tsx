import React from "react";
const heartStyle = {
  fill: "rgba(255, 0, 0, 0.3)",
  width: "20px",
  marginRight: "10px"
};
const full = {
  fill: "red"
};

function Heart(props: any) {
  return (
    <>
      <svg
        style={props.class ? { ...heartStyle, ...full } : heartStyle}
        viewBox="0 0 32 29.6"
      >
        <path
          d="M23.6,0c-3.4,0-6.3,2.7-7.6,5.6C14.7,2.7,11.8,0,8.4,0C3.8,0,0,3.8,0,8.4c0,9.4,9.5,11.9,16,21.2
        c6.1-9.3,16-12.1,16-21.2C32,3.8,28.2,0,23.6,0z"
        />
      </svg>
    </>
  );
}

export default Heart;
