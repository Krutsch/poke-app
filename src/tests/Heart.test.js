import React from "react";
import Heart from "../components/Heart";
import ShallowRenderer from "react-test-renderer/shallow";

it("renders correctly", () => {
  const renderer = new ShallowRenderer();
  renderer.render(<Heart />);
  const result = renderer.getRenderOutput();
  expect(result).toMatchSnapshot();
});
