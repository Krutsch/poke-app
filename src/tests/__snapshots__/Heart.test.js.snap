// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<React.Fragment>
  <svg
    style={
      Object {
        "fill": "rgba(255, 0, 0, 0.3)",
        "marginRight": "10px",
        "transition": "fill 0.1s ease",
        "width": "20px",
      }
    }
    viewBox="0 0 32 29.6"
  >
    <path
      d="M23.6,0c-3.4,0-6.3,2.7-7.6,5.6C14.7,2.7,11.8,0,8.4,0C3.8,0,0,3.8,0,8.4c0,9.4,9.5,11.9,16,21.2
 c6.1-9.3,16-12.1,16-21.2C32,3.8,28.2,0,23.6,0z"
    />
  </svg>
</React.Fragment>
`;
